### PR TITLE
PE-318: list drive command

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

File: src/arfsdao.ts [hunk 1]
```diff
@@ -1113,6 +1115,7 @@ export class FolderHierarchy {
 	}
 
 	public entityPathToFolderId(folderId: FolderID): string {
+		if (folderId === 'root folder') return '/';
```
Comment (matibat) [inline src/arfsdao.ts:1118]: Remove this
```suggestion
```

File: src/commands/list_folder.ts [hunk 2]
```diff
@@ -36,14 +32,14 @@ new CLICommand({
 			const arDrive = new ArDrive(
 				wallet,
 				cliWalletDao,
-				new ArFSDAO(wallet, arweave),
-				new ArDriveCommunityOracle(arweave),
+				new ArFSDAO(wallet, cliArweave),
+				new ArDriveCommunityOracle(cliArweave),
```
Comment (matibat) [inline src/commands/list_folder.ts:36]: Shouldn't we be getting these from injection? Maybe move it to the context class???

Comment (matibat) [reply src/commands/list_folder.ts:36]: For this question, I think you are the guy @arielmelendez! ;)

Comment (matibat) [reply src/commands/list_folder.ts]: Outdated. Closing.

File: src/arfsdao.ts [hunk 3]
```diff
@@ -244,7 +246,11 @@ export class ArFSDAOAnonymous extends ArFSDAOType {
 	 * @param {FolderID} folderId the folder ID to list children of
 	 * @returns {ArFSPublicFileOrFolderWithPaths[]} an array representation of the children and parent folder
 	 */
-	async listPublicFolder(folderId: FolderID, maxDepth = 0): Promise<ArFSPublicFileOrFolderWithPaths[]> {
+	async listPublicFolder(
+		folderId: FolderID,
+		maxDepth: number,
+		includeRoot: boolean
+	): Promise<ArFSPublicFileOrFolderWithPaths[]> {
```
Comment (arielmelendez) [inline src/arfsdao.ts:253]: • Add validation for maxDepth value (>= 0).
• Update docblock

File: src/arfsdao.ts [hunk 4]
```diff
@@ -747,7 +757,8 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 	async listPrivateFolder(
 		folderId: FolderID,
 		driveKey: DriveKey,
-		maxDepth = 0
+		maxDepth: number,
```
Comment (arielmelendez) [inline src/arfsdao.ts:760]: Validate max depth and update docblocks.

File: src/commands/list_drive.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import { alphabeticalOrder } from '../utils/sort_functions';
+
+new CLICommand({
+	name: 'list-drive',
+	parameters: [DriveIdParameter, SeedPhraseParameter, WalletFileParameter],
+	async action(options) {
+		const parameters = new ParametersHelper(options, cliWalletDao);
+		const driveId = parameters.getRequiredParameterValue(DriveIdParameter);
+		let children: (ArFSPrivateFileOrFolderWithPaths | ArFSPublicFileOrFolderWithPaths)[];
+		const maxDepth = 0;
```
Comment (arielmelendez) [inline src/commands/list_drive.ts:16]: Why is the default behavior to list a max depth of 0? As per the manpage:
```
list-drive:
  • Gets all ArFS Metadata and Txids for the files and folders within this drive
  --drive-id: the ArFS drive ID for the drive
  --drive-password: the drive password for drive of the folder identified by --drive-id
  • Required only for folders residing in private drives
  • Can NOT be used in conjunction with --drive-key
  --drive-key: the drive key for the parent drive of the folder identified by --folder-id
  • Required only for folders residing in private drives
  • Can NOT be used in conjunction with --drive-password
  --get-all-revisions: (OPTIONAL) gets every revision of each ArFS entity within this drive
```
most notably `Gets all ArFS Metadata and Txids for the files and folders within this drive`, key word `all`. So I believe the max depth value here should be Number.MAX_SAFE_INTEGER (or Number.POSITIVE_INFINITY).

File: src/commands/list_drive.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import { arDriveFactory, cliArweave, cliWalletDao } from '..';
+import { ArDriveAnonymous } from '../ardrive';
+import { ArFSDAOAnonymous, ArFSPrivateFileOrFolderWithPaths, ArFSPublicFileOrFolderWithPaths } from '../arfsdao';
+import { CLICommand, ParametersHelper } from '../CLICommand';
+import { DriveIdParameter, SeedPhraseParameter, WalletFileParameter } from '../parameter_declarations';
+import { alphabeticalOrder } from '../utils/sort_functions';
+
+new CLICommand({
+	name: 'list-drive',
+	parameters: [DriveIdParameter, SeedPhraseParameter, WalletFileParameter],
```
Comment (arielmelendez) [inline src/commands/list_drive.ts:11]: You're missing the drive password and drive key parameters for private drives. This doesn't work for private drives.

**Review by arielmelendez — CHANGES_REQUESTED** [review]
This code needs to be tested. It's not working for me locally.

File: src/commands/list_drive.ts (hunk 6 above)
Comment (arielmelendez) [reply src/commands/list_drive.ts]: c4e2a90

File: src/arfsdao.ts (hunk 4 above)
Comment (arielmelendez) [reply src/arfsdao.ts:769]: eee1dba

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**